File: src/interfaces.php
<?php

interface Bovino
{
    public function comer(string $comida, int $kilos);
    public function caminar(float $velocidad, float $distancia);
}

final class Vaca implements Bovino
{
    public function comer(string $comida, int $kilos) {
        print("Vaca comiendo $kilos kilos de $comida" . PHP_EOL);
    }

    public function caminar(float $velocidad, float $distancia) {
        print("Vaca caminando $distancia Km a $velocidad Km/h" . PHP_EOL);
    }
}

class Humano
{
    public function comer(string $comida, float $kilos) {
        print("Humano comiendo $kilos kilos de $comida" . PHP_EOL);
    }

    public function caminar(float $velocidad, float $distancia) {
        print("Humano caminando $distancia Km a $velocidad Km/h" . PHP_EOL);
    }
}

function ingresarAlCampo(Bovino $animal) {
    $animal->comer("pasto", 5);
    $animal->caminar(30.1, 3.2);
}

$vaca = new Vaca();
ingresarAlCampo($vaca);

$humano = new Humano();
ingresarAlCampo($humano);

File: src/interfaces.py

from typing import Protocol

class Bovino(Protocol):
    def comer(self, comida: str, kilos: int):
        ...

    def caminar(self, velocidad: float, distancia: float):
        ...


class Vaca:
    def comer(self, comida: str, kilos: int):
        print(f"Vaca comiendo {kilos} kilos de {comida}")

    def caminar(self, velocidad: float, distancia: float):
        print(f"Vaca caminando {distancia} Km a {velocidad} Km/h")


class Humano:
    def comer(self, comida: str, kilos: int):
        print(f"Humano comeiendo {kilos} kilos de {comida}")

    def caminar(self, velocidad: float, distancia: float):
        print(f"Humano caminando {distancia} Km a {velocidad} Km/h")


def ingresar_a_campo(animal: Bovino):
    animal.comer("pasto", 5);
    animal.caminar(30.1, 3.2);


vaca = Vaca()
ingresar_a_campo(vaca)
humano = Humano()
ingresar_a_campo(humano)

